Use tensor's quantized properties directly in pickler

iseeyuan opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#46267 Use tensor's quantized properties directly in pickler**

When serializing quantized model, the quantized info could be directly accessed from the tensor itself, instead of going around by calling Quantizer. Calling Quantizer introduced some un-necessary coupling between the serializer and ATen/native. 

Differential Revision: [D24283008](https://our.internmc.facebook.com/intern/diff/D24283008)

iseeyuan committed:
Use tensor's quantized properties directly in pickler

[ghstack-poisoned]

jerryzh168 commented on torch/csrc/jit/serialization/pickler.cpp:
```diff
@@ -395,9 +395,10 @@ void Pickler::pushLiteralTensor(const IValue& ivalue) {
       case at::kPerChannelAffine: {
         const auto* quantizer = static_cast<at::PerChannelAffineQuantizer*>(
```
this can be removed as well

jerryzh168 reviewed (approved):
thanks!

iseeyuan committed:
Update on "Use tensor's quantized properties directly in pickler"


When serializing quantized model, the quantized info could be directly accessed from the tensor itself, instead of going around by calling Quantizer. Calling Quantizer introduced some un-necessary coupling between the serializer and ATen/native. 

Differential Revision: [D24283008](https://our.internmc.facebook.com/intern/diff/D24283008)

[ghstack-poisoned]

iseeyuan committed:
Update on "Use tensor's quantized properties directly in pickler"


When serializing quantized model, the quantized info could be directly accessed from the tensor itself, instead of going around by calling Quantizer. Calling Quantizer introduced some un-necessary coupling between the serializer and ATen/native. 

Differential Revision: [D24283008](https://our.internmc.facebook.com/intern/diff/D24283008)

[ghstack-poisoned]

iseeyuan committed:
Update on "Use tensor's quantized properties directly in pickler"


When serializing quantized model, the quantized info could be directly accessed from the tensor itself, instead of going around by calling Quantizer. Calling Quantizer introduced some un-necessary coupling between the serializer and ATen/native. 

Differential Revision: [D24283008](https://our.internmc.facebook.com/intern/diff/D24283008)

[ghstack-poisoned]

iseeyuan committed:
Update on "Use tensor's quantized properties directly in pickler"


When serializing quantized model, the quantized info could be directly accessed from the tensor itself, instead of going around by calling Quantizer. Calling Quantizer introduced some un-necessary coupling between the serializer and ATen/native. 

Differential Revision: [D24283008](https://our.internmc.facebook.com/intern/diff/D24283008)

[ghstack-poisoned]
